lib/collection/src/operations/verification/query.rs

```diff
@@ -89,8 +145,14 @@ impl StrictModeVerification for CollectionPrefetch {
         }
 
         if let Some(query) = self.query.as_ref() {
+            // check if prefetch can perform a fullscan
+            query
+                .check_fullscan(&self.using, collection, strict_mode_config)
+                .await?;
             // check for unindexed fields in formula
-            query.check_strict_mode(collection, strict_mode_config)?
+            query
+                .check_strict_mode(collection, strict_mode_config)
```

Comment:
Function naming in this case becomes confusing.
`check_fullscan` is followed by `check_strict_mode`. Isn't `check_fullscan` part of strict mode?

Reply:
`check_strict_mode` on the `Query` does not have the knowledge whether there are prefetches.
This information is necessary to know if it is a rescoring use case.

For `CollectionPrefetch` it should always be checked.
But for `CollectionQueryRequest` only if there are no prefetches.

This is why the calls are not unified.